File: app/src/main/java/com/elegion/android/template/ui/features/adapter/FeaturesAdapter.java
package com.elegion.android.template.ui.features.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.elegion.android.template.R;
import com.elegion.android.template.data.model.Feature;
import com.elegion.android.template.ui.base.adapter.AbstractPaginationAdapter;
import com.elegion.android.template.ui.features.adapter.holder.FeatureViewHolder;

import java.util.ArrayList;
import java.util.List;

/**
 * @author mikhail.barannikov on 24.07.2017
 */

public class FeaturesAdapter extends AbstractPaginationAdapter<FeatureViewHolder> {
    private final List<Feature> mItems = new ArrayList<>();

    @Override
    public FeatureViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        final LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        final View view = inflater.inflate(R.layout.li_feature, parent, false);
        return new FeatureViewHolder(view);
    }

    @Override
    public void onBindViewHolder(FeatureViewHolder holder, int position) {
        super.onBindViewHolder(holder, position);
        holder.bind(mItems.get(position));
    }

    @Override
    public int getItemCount() {
        return mItems.size();
    }

    public void setData(List<Feature> items, boolean clear) {
        if (clear) {
            mItems.clear();
        }
        mItems.addAll(items);
        notifyDataSetChanged();
    }
}

File: app/src/main/java/com/elegion/android/template/ui/features/adapter/FeaturesAdapter.kt
package com.elegion.android.template.ui.features.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import com.elegion.android.template.R
import com.elegion.android.template.data.model.Feature
import com.elegion.android.template.ui.base.adapter.AbstractPaginationAdapter
import com.elegion.android.template.ui.features.adapter.holder.FeatureViewHolder
import java.util.*

class FeaturesAdapter : AbstractPaginationAdapter<FeatureViewHolder>() {
    private val mItems = ArrayList<Feature>()

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): FeatureViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val view = inflater.inflate(R.layout.li_feature, parent, false)
        return FeatureViewHolder(view)
    }

    override fun onBindViewHolder(holder: FeatureViewHolder, position: Int) {
        super.onBindViewHolder(holder, position)
        holder.bind(mItems[position])
    }

    override fun getItemCount(): Int = mItems.size

    fun setData(items: List<Feature>, clear: Boolean) {
        if (clear) {
            mItems.clear()
        }
        mItems.addAll(items)
        notifyDataSetChanged()
    }
}
